corehq/apps/locations/forms.py

```diff
@@ -632,15 +632,28 @@ class RelatedLocationForm(forms.Form):
 
     def __init__(self, domain, location, *args, **kwargs):
         self.location = location
-        kwargs['initial'] = {
-            'related_locations': ','.join(LocationRelation.from_locations([self.location]))
-        }
+        self.related_location_ids = LocationRelation.from_locations([self.location])
+        kwargs['initial'] = {'related_locations': ','.join(self.related_location_ids)}
         super(RelatedLocationForm, self).__init__(*args, **kwargs)
 
         self.fields['related_locations'].widget = LocationSelectWidget(
             domain, id='id_related_locations', multiselect=True
         )
 
+        locations = (
+            SQLLocation.objects
+            .filter(location_id__in=self.related_location_ids)
+        )
+
+        distance_dict = LocationRelation.relation_distance_dictionary([self.location])
+
+        for loc in locations:
+            distance = distance_dict[loc.location_id].get(self.location.location_id)
```

Comment:
I think a simpler way to do this would be to pull it up front when you look up the `LocationRelation`s - that is, get all related locations and their distances together.

Reply:
That makes sense. I will look at simplifying this a bit in a follow up PR.